### PR TITLE
feat(cb2-9598): validation for dates

### PR DESCRIPTION
**Pull request by kpearmanbjss**
## Validation for dates - not in the future

_Adds validation to most date fields on technical records_
[CB2-9598](https://dvsa.atlassian.net/browse/CB2-9598)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/utils/error-message-map.spec.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ describe('ErrorMessageMap', () => {
     ['Date must be in the future', ValidatorNames.FutureDate, [true, 'Date']],
     ['This date must be in the future', ValidatorNames.FutureDate, [true, undefined]],
     ['This date must be ahead of the previous date', ValidatorNames.AheadOfDate, [true, undefined]],
+    ['This year must be this current year or in the past', ValidatorNames.PastYear, [true, undefined]],
```
Comment (Twebb97) [inline src/app/forms/utils/error-message-map.spec.ts:33]: Is this gramatically correct ?

File: src/app/forms/validators/custom-validators.spec.ts [hunk 2]
```diff
@@ -396,6 +396,24 @@ describe('futureDate', () => {
...
+describe('pastYear', () => {
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('2022-01-01T00:00:00.000Z'));
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  it.each([
```
Comment (Twebb97) [inline src/app/forms/validators/custom-validators.spec.ts:408]: Love this

**Comment by Twebb97** [comment]
Ignore some of these changes, just checking what some of these buttons do locally

**Comment by kpearmanbjss** [comment]
Still waiting on confirmation from Nick about a field that may need to be included as well as wording of error, just wanted to set the PR up incase it was fine as is.

**Review by pbardy2000 — APPROVED** [review]

**Review by owen-corrigan-bjss — APPROVED** [review]

**Pull request merged**